src/plugins/hetero/src/compiled_model.cpp

```diff
@@ -37,14 +37,6 @@ ov::hetero::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model
 }
 
 void ov::hetero::CompiledModel::compile_model(const std::shared_ptr<ov::Model>& model) {
-    // Calling of ConstantFolding in HETERO plugin is required because
-    // in some cases topology split is happening after constant subgraph.
-    // It may cause replacement of Constant by Parameter in such operations
-    // like Reshape/Transpose/Gather and lead to unexpected dynamism or exception
-    ov::pass::Manager manager;
```

Comment:
can we test a graph described here? for example, transpose node requires the input node (1) to be constant:
src/plugins/intel_gpu/src/plugin/ops/transpose.cpp
if so, maybe we need another PR to cover situation like this, instead of do early constant folding in hetero.

Reply:
in general, I think this PR is doing the right thing not to introduce CF in hetero, but, we need to understand the exceptions and create JIRA to follow

Reply:
Hi, @ilya-lavrenov @songbell , added more test case to cover this situation, please review, thanks.